### ISSUE TITLE
SMAC tuner with multiple trial concurrency

### ISSUE THREAD

**Issue by choant** [microsoft/nni#4960]
**Describe the issue**:
Hello!
I have been using nni for my project for a month and now I am trying to use multiple remote machines( 8 nodes with 8GPU) with trial concurrency set 8.

From what I've understood, SMAC tuner uses previous results to make surrogate model for hyperparameters. However, since there are 8 experiments running simultatneously on different nodes, I'm not so sure whether this surrogate model constructed by results obtained from all 8 nodes.

I think it either works as 
a) The sinlge and only one SMAC uses hyperparameter and metric from finished trials of all 8 nodes to build surrogate model.
or
b) 8 SMAC tuners are constructed and each one uses hyperparameter and metric from 8 nodes, respectively.

Could you let me know how it is worked(a or b or in some different way)? 

Thanks for your time!

**Environment**:
- NNI version: 2.7
- Training service (local|remote|pai|aml|etc): remote
- Client OS: Linux
- Server OS (for remote mode only): Linux
- Python version: python 3.8.5
- PyTorch/TensorFlow version: pytorch 1.10
- Is conda/virtualenv/venv used?: Yes
- Is running in Docker?: NO

**Configuration**:

experimentName: myexperiment
searchSpace:
   my search space
trialCommand: my command
trialCodeDirectory: 
trialGpuNumber: 8 

trialConcurrency: 8             
maxTrialNumber: myTrial     
maxExperimentDuration: myduration
   
assessor:
  name: Medianstop
  classArgs:
    optimize_mode: minimize
    start_step : my_start
tuner:                          
  name: SMAC             
  classArgs:                    
    optimize_mode: minimize
trainingService:

  platform: remote
  machinelist:
            - host: myhost1
            - host: myhost2
            - host: myhost3
            - host: myhost4
            - host: myhost5
            - host: myhost6
            - host :myhost7
            - host: myhost8

**Log message**:
 - nnimanager.log:
 - dispatcher.log:
 - nnictl stdout and stderr:
 

**How to reproduce it?**:

**Comment by scarlett2018** [comment]
> hi @choant , it is option a)

@choant - I'm closing the issue as the question has been answered, feel free to reopen the issue if you need more discussions or clarifications. Thanks!
